### ISSUE TITLE
2-in-1 shared heater

### ISSUE THREAD

**Issue by cbldwn** [Ultimaker/Cura#8255]
Hello,
   I'm running a y-splitter in my bowden tube to my hotend and had been using the shared heater option in the printer settings, which in the new 4.7 looks to be removed. This needs to be added back as every time the hotend does a tool change its dropping  the temperature to 195 even tho I have the standby temp set at 212. 

Thanks,
     Craig Baldwin

**Comment by fieldOfView** [comment]
As far as I can see, the option is still there. Perhaps I could reproduce your issue if you had filled in the issue template.

**Comment by cbldwn** [comment]
It is not still there, there isn’t even a box to check for it.
![DB739565-5EBE-4546-85F2-6D65DFB8045E](https://user-images.githubusercontent.com/68358507/91462164-1ae8ae00-e847-11ea-82be-505484a7aa0a.jpeg)

**Comment by fieldOfView** [comment]
I am sorry, you are right. The machine I checked with was still running Cura 4.7 beta. It does seem the feature has been temporarily removed.

The "Shared Heater" feature is temporarily disabled because its implementation is incomplete. Printers with multiple filaments going into one nozzle will keep the inactive filaments retracted at the start of a print. However CuraEngine assumes that all filaments start at the nozzle tip. So it'll start printing the second filament without unretracting it.

See: https://github.com/Ultimaker/Cura/issues/8148

**Comment by cbldwn** [comment]
Yes that’s not a problem, don’t use cura’s tool change. Put the filament load command in the tool start and the retract in the end code and then everytime Cura calls a tool it will load the filament before before continuing to print and unload the filament before loading the next extruder. I’ve have been using this setup without any problems until the update.

**Comment by fieldOfView** [comment]
If you don't mind hacking Cura a bit, you could replace `C:\Program Files\Ultimaker Cura 4.7\plugins\MachineSettingsAction\MachineSettingsPrinterTab.qml` with this version: https://raw.githubusercontent.com/Ultimaker/Cura/c7e6553dbfb5cdcfbe0ba0e7b263d64062fe0952/plugins/MachineSettingsAction/MachineSettingsPrinterTab.qml

**Comment by cbldwn** [comment]
I don’t mind, I’ve already written my own .json files for my printer to make upgrading quicker and easier for me. What is more of an issue with the prime tower is the order in which the prime tower prints when I was setting up my 2-in-1-out splitter and trying to get it working I noticed a lot of times after a tool change instead of printing the purge tower it would start printing the part. And then go to the purge tower. Because of this I went away from the purge tower and started using a purge bucket, not only did it eliminate my issue of printing the part before purging but it also dropped my print time from 45 min to 25. The black and gold one was printed with the prime tower and the gold and copper with my purge bucket for comparison.

![F975F7C1-9F67-4A33-A55E-E7DE10FC10B7](https://user-images.githubusercontent.com/68358507/91465271-d232f400-e84a-11ea-9577-e78a76ed46b5.jpeg)
![DB3C6A81-77F1-49BD-8054-A4D4CA690402](https://user-images.githubusercontent.com/68358507/91465275-d2cb8a80-e84a-11ea-8650-ad7ad2fb0c79.jpeg)

**Comment by fieldOfView** [comment]
That sounds like another issue. Issues here should be one specific problem. It seems your original problem is solved for now, so you can close this issue. 

I just felt bad about saying the checkbox was still there while it wasn't, so I wanted to give you a workaround. I know very little about shared heaters and prime towers otherwise.

**Comment by cbldwn** [comment]
The issues I mentioned with the prime tower would probably part of the reason why this was disabled in the first place. Another work around to the original filament un-retract problem is suppose to be if running marlin, to enable firmware retract and have Cura output the g10 & g11 for normal retraction and let the firmware handle the retraction for the tool changes. Thanks for the information on where I would have to make the change. I was able to skim through the code and found the comment I had to move up to re-enable the shared heater check box. Photo’s of the change are attached so if someone else has the problem and wants to make the change back as well. 
![5FA1FEC0-2C11-4371-A419-6AAB88FE2066](https://user-images.githubusercontent.com/68358507/91467422-c98fed00-e84d-11ea-95ca-b1623c2b2bd7.jpeg)
![3F662092-B82D-415D-8B0F-1E41ADDFB3DE](https://user-images.githubusercontent.com/68358507/91467425-cac11a00-e84d-11ea-8ce4-cbd23e7db09b.jpeg)
